### PR TITLE
Feat: Upcoming events functionality

### PR DESCRIPTION
**Pull request by kmaid**

### CODE REVIEW

**Review by fampinheiro — APPROVED** [review]

File: apps/frontend/src/events/Upcoming.tsx [hunk 1]
```diff
@@ -0,0 +1,39 @@
+import React from 'react';
+import { EventsUpcoming } from '@asap-hub/react-components';
+
+import { useEvents } from './state';
+import { usePagination, usePaginationParams } from '../hooks';
+import { NETWORK_PATH, EVENTS_PATH } from '../routes';
+import { GROUPS_PATH } from '../network/routes';
+
+type UpcomingProps = {
+  time: string;
```
Comment (jeysal) [inline apps/frontend/src/events/Upcoming.tsx:10]:
```suggestion
  readonly time: string;
```

File: apps/frontend/src/events/Events.tsx [hunk 2]
```diff
@@ -1,27 +1,45 @@
...
+import { EventsPage } from '@asap-hub/react-components';
+import { Switch, Route, useRouteMatch, Redirect } from 'react-router-dom';
 
 import Frame from '../structure/Frame';
 import Event from './Event';
-import { useCalendars } from './calendar/state';
+import { EVENTS_UPCOMING_PATH, EVENTS_CALENDAR_PATH } from './routes';
+
+const loadCalendars = () =>
+  import(/* webpackChunkName: "events-calendar" */ './calendar/Calendars');
```
Comment (jeysal) [inline apps/frontend/src/events/Events.tsx:10]:
```suggestion
  import(/* webpackChunkName: "events-calendars" */ './calendar/Calendars');
```

File: apps/frontend/src/events/state.ts [hunk 3]
```diff
@@ -1,9 +1,58 @@
...
+import { EventResponse, ListEventResponse } from '@asap-hub/model';
 
 import { authorizationState } from '../auth/state';
-import { getEvent } from './api';
+import { getEvent, getEvents, BeforeOrAfter } from './api';
+import { GetListOptions } from '../api-util';
 
+const eventIndexState = atomFamily<
+  { ids: ReadonlyArray<string>; total: number } | Error | undefined,
+  GetListOptions
```
Comment (jeysal) [inline apps/frontend/src/events/state.ts:17]: `GetListOptions & BeforeOrAfter`, right?

File: apps/frontend/src/events/state.ts [hunk 4]
```diff
@@ -1,9 +1,58 @@
...
+      if (event === undefined) return undefined;
+      events.push(event);
+    }
+    return { total: index.total, items: events };
+  },
+  set: (options) => ({ get, set, reset }, newEvents) => {
+    if (newEvents === undefined || newEvents instanceof DefaultValue) {
+      const oldGroups = get(eventIndexState(options));
+      if (!(oldGroups instanceof Error)) {
+        oldGroups?.ids?.forEach((id) => reset(eventState(id)));
```
Comment (jeysal) [inline apps/frontend/src/events/state.ts:42]:
```suggestion
      const oldEvents = get(eventIndexState(options));
      if (!(oldEvents instanceof Error)) {
        oldEvents?.ids?.forEach((id) => reset(eventState(id)));
```

**Review by jeysal — CHANGES_REQUESTED** [review]
mostly lgtm just details

**Comment by jeysal** [comment]
pushed a small fix that hopefully avoids the missing cov. it's not really pretty in general with resetting these recoil families though by guessing the family arguments

**Comment by jeysal** [comment]
If you're fine with these suggested changes commit them and I think we can land

**Comment by jeysal** [comment]
Design is not gonna like the 'only event title clickable' thing though are they? :D

**Review by jeysal — APPROVED** [review]
Without istanbul-ignore :tada: honestly I would have rather overridden the coverage check than put istanbul ignore on a legitimate and essential production code file :sweat_smile:

**Pull request merged**